### PR TITLE
Small Click refactor.

### LINKED ISSUE #25814: Two people being able to pick up the same item
Server revision compiled on: 2017-04-02
The following pull requests are currently test merged:
#25801: '[DNM]Revert to last server compile' by Cyberboss 

Based off master commit:
6405a412a1122776f620ea5d142c8141beb11945 

I was going to write it off as a goof but it happened at least 5 times over the last week or so.
Sometimes when two people are fighting over an object they can pick up at the same time which does one or more of the following:
The item is inhand for one person but invisible and they can't pick up anything else on that slot.
The item is undroppable or duplicates on drop.
One case the mime was throwing a toolbox we both picked up, and it was animating(doing the rotating thing) in hand.
 Some other weird behaviours I don't remember exactly.

### LINKED ISSUE #26796: Reach weapon issues
@Xhuis 

- When you attack somebody with a reach weapon, an invisible "obj" item appears under them.
obj

- You can't attack prone people with reach weapons for some reason.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #25814 
Fixes #26796

Is this better ? I'm honestly not sure.
The cost for deep inventory click is now higher, GetAllContents might get pricey.
For now i didn't change the click validation checks for what in contents should be reachable to preserve old functionality but this should probably be changed to only allow inventory access.

### CODE REVIEW

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -107,41 +107,84 @@
...
+	return FALSE
+
+//Can [target] in this container be reached by [user], can't be more than [depth] levels deep
+/atom/proc/CanReachStorage(atom/target,user,depth)
+	return FALSE
+
+/obj/item/weapon/storage/CanReachStorage(atom/target,user,depth)
+	while(target && depth > 0)
+		target = target.loc
+		depth --
```
Comment (RemieRichards) [inline code/_onclick/click.dm]: dont space this operator.

**Comment by PKPenguin321** [comment]
Fixes https://github.com/tgstation/tgstation/issues/26796 ?

**Comment by AnturK** [comment]
Partially, i'll add fix for the other part in a moment.

**Comment by AnturK** [comment]
Some circular calls seem to be happening. Gotta track it down.

**Comment by Razharas** [comment]
Another solved case

**Pull request merged**